server/server.go

```diff
@@ -9,14 +9,28 @@ import (
 	"net/url"
 	"os"
 	"strings"
+	"sync"
 
 	"github.com/google/uuid"
 	"github.com/gorilla/websocket"
 )
 
-var connections = map[UserId]*Conn{}
-var users = map[UserId]UserName{}
-var games = map[GameId]*Game{}
+type globalState struct {
+	*sync.Mutex
+	connections map[UserId]*Conn
+	users       map[UserId]UserName
+	games       map[GameId]*Game
+}
+
+// state is this server's global state. It is coarsely locked by
```

Comment:
"Coarsely" here indicates that we lock all of state at a time and not one attribute at a time, yes?